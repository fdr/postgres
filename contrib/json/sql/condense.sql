SELECT json('"hello"');
SELECT json($$"hello\u266Bworld"$$);
SELECT json($$"hello\u266bworld"$$);
SELECT json($$"helloâ™«world"$$);
SELECT json($$      "hello world"    $$);
SELECT json($$    {  "hello" : "world"}    $$);
SELECT json($$    {  "hello" : "world", "bye": 0.0001  }    $$);
SELECT json($$    {  "hello" : "world",
								"bye": 0.0000001
}    $$);
SELECT json($$    {  "hello" : "world"
,
"bye"
: [-0.1234e1, 12345e0]		}    $$);
SELECT json($$"\u007E\u007F\u0080"$$)::text = E'\"\u007E\u007F\u0080\"';
SELECT json($$"\u00FE\u00FF\u0100"$$);
SELECT json($$"\uD835\uDD4E"$$);
SELECT json($$"\uD835\uD835"$$);

-- These should be turned into single-character escapes.
SELECT json($$"\u0022"$$)::text = $$"\""$$;
SELECT json($$"\u005c"$$)::text = $$"\\"$$;
SELECT json($$"\u0008"$$)::text = $$"\b"$$;
SELECT json($$"\u000c"$$)::text = $$"\f"$$;
SELECT json($$"\u000a"$$)::text = $$"\n"$$;
SELECT json($$"\u000d"$$)::text = $$"\r"$$;
SELECT json($$"\u0009"$$)::text = $$"\t"$$;
