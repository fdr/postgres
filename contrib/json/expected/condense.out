SELECT json('"hello"');
  json   
---------
 "hello"
(1 row)

SELECT json($$"hello\u266Bworld"$$);
     json      
---------------
 "hello‚ô´world"
(1 row)

SELECT json($$"hello\u266bworld"$$);
     json      
---------------
 "hello‚ô´world"
(1 row)

SELECT json($$"hello‚ô´world"$$);
     json      
---------------
 "hello‚ô´world"
(1 row)

SELECT json($$      "hello world"    $$);
     json      
---------------
 "hello world"
(1 row)

SELECT json($$    {  "hello" : "world"}    $$);
       json        
-------------------
 {"hello":"world"}
(1 row)

SELECT json($$    {  "hello" : "world", "bye": 0.0001  }    $$);
              json              
--------------------------------
 {"hello":"world","bye":0.0001}
(1 row)

SELECT json($$    {  "hello" : "world",
								"bye": 0.0000001
}    $$);
               json                
-----------------------------------
 {"hello":"world","bye":0.0000001}
(1 row)

SELECT json($$    {  "hello" : "world"
,
"bye"
: [-0.1234e1, 12345e0]		}    $$);
                    json                     
---------------------------------------------
 {"hello":"world","bye":[-0.1234e1,12345e0]}
(1 row)

SELECT json($$"\u007E\u007F\u0080"$$)::text = E'\"\u007E\u007F\u0080\"';
 ?column? 
----------
 t
(1 row)

SELECT json($$"\u00FE\u00FF\u0100"$$);
 json  
-------
 "√æ√øƒÄ"
(1 row)

SELECT json($$"\uD835\uDD4E"$$);
 json 
------
 "ùïé"
(1 row)

SELECT json($$"\uD835\uD835"$$);
      json      
----------------
 "\uD835\uD835"
(1 row)

-- These should be turned into single-character escapes.
SELECT json($$"\u0022"$$)::text = $$"\""$$;
 ?column? 
----------
 t
(1 row)

SELECT json($$"\u005c"$$)::text = $$"\\"$$;
 ?column? 
----------
 t
(1 row)

SELECT json($$"\u0008"$$)::text = $$"\b"$$;
 ?column? 
----------
 t
(1 row)

SELECT json($$"\u000c"$$)::text = $$"\f"$$;
 ?column? 
----------
 t
(1 row)

SELECT json($$"\u000a"$$)::text = $$"\n"$$;
 ?column? 
----------
 t
(1 row)

SELECT json($$"\u000d"$$)::text = $$"\r"$$;
 ?column? 
----------
 t
(1 row)

SELECT json($$"\u0009"$$)::text = $$"\t"$$;
 ?column? 
----------
 t
(1 row)

